### ISSUE TITLE
Inkbird ISV-200W

### ISSUE THREAD

**Issue by vadimbz** [tuya/tuya-home-assistant#705]
**Device normal info**
- Inkbird
- ISV-200W

I'm sorry, but my output seems to be less detailed than the example. I'll do my best to provide additional information, should it be needed. Thanks in advance!

**Device info 
{
"active_time": 1637194452,
"biz_type": 0,
"category": "mzj",
"create_time": 1629615339,
"icon": "smart/icon/1557470235chhak9ddel5_0.png",
"id": "25087437f4cfa2ddd1e4",
"ip": "***",
"lat": "***",
"local_key": "***",
"lon": "***",
"model": "",
"name": "Sous Vide",
"online": false,
"owner_id": "55126974",
"product_id": "iPiqkMPG2lOUtpqB",
"product_name": "Sous Vide",
"status": [],
"sub": false,
"time_zone": "+07:00",
"uid": "***",
"update_time": 1637197955,
"uuid": "25087437f4cfa2ddd1e4"
}

**Device specifications:**
2021-11-18 18:11:28 DEBUG (SyncWorker_6) [tuya_iot] Request: method = GET, url = https://openapi.tuyaus.com/v1.0/devices/25087437f4cfa2ddd1e4/specifications, params = None, body = None, t = 1637233888628

**Comment by frenck** [comment]
`mzj` is not a supported device category at this moment, thus not a bug, but a feature request.

We use GitHub for tracking issues, not for providing support or tracking feature requests.

If you want to suggest a feature, you should try our [Community Forum: Feature Requests](https://community.home-assistant.io/c/feature-requests).

If you have additional questions, feel free to join our [Discord chat server](https://discord.gg/c5DvZ4e).

Thanks! 👍

**Comment by vadimbz** [comment]
> `mzj` is not a supported device category at this moment, thus not a bug, but a feature request.
> 
> If you want to suggest a feature, you should try our [Community Forum: Feature Requests](https://community.home-assistant.io/c/feature-requests).
> 
> I think that is a better place than here, as this repository doesn't maintain an integration anymore.
I opened new device support request, and followed the template (data from the log). Also, it's marked as feature request/enhancement/help wanted.
But sure, I'll create a matching forum thread.

**Comment by vadimbz** [comment]
[Feature Request on HA Forum](https://community.home-assistant.io/t/integrating-tuya-powered-sous-vide-machines-mzj-category-inkbird-isv-200w/357977)

**Comment by vadimbz** [comment]
Here's data from another Tuya-powered Sous Vide owner:
{
      "active_time": 1637278418,
      "biz_type": 18,
      "category": "mzj",
      "create_time": 1637276422,
      "icon": "smart/icon/ay1529490557486Rz8BL/15614298450018d08e946.jpg",
      "id": "186510413c6105cbcf77",
      "ip": "***",
      "lat": "***",
      "local_key": "***",
      "lon": "***",
      "model": "KASMSVDC13A",
      "name": " Sous Vide",
      "online": true,
      "owner_id": "46208002",
      "product_id": "Kj07pTFP31ATT1in",
      "product_name": " Sous Vide",
      "status": [],
      "sub": false,
      "time_zone": "+12:00",
      "uid": "***",
      "update_time": 1637632392,
      "uuid": "186510413c6105cbcf77"
    }

**Comment by frenck** [comment]
Yesterday, Home Assistant 2022.2 was released and it contains a new diagnostic tool.

There is now a "Download Diagnostics" button on each Tuya device on the device page. Clicking that button will download diagnostics information for that specific device. Drag the downloaded file into this issue.

This will help us learn about your device and provide details for implementing/adding support.

Thanks!

**Comment by Zarkstar** [comment]
Hello. I have the ISV-100W, which is the 1000W version of ISV-200W, but it should be practically the same. I've attached the diagnostics file. Hope it will help!

[tuya-4b3c280461170972843ad005306ffbb4-Sous Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json.txt](https://github.com/tuya/tuya-home-assistant/files/8018219/tuya-4b3c280461170972843ad005306ffbb4-Sous.Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json.txt)
s

**Comment by frenck** [comment]
Thanks for your diagnostics @Zarkstar

```json
    "function": {},
    "status_range": {},
    "status": {},
```

☝️  That is what the Tuya provides about the available functions and statuses for your device. As you can see, there is no data in there. So, there is nothing for us to integrate @ Home Assistant 😟

**Comment by Zarkstar** [comment]
*Queue sad music* Thank you for taking the time to explain how the diagnostics works.
Is that something that may change in the future (aka should I keep that diagnostic monitored)? The Tuya IoT Platform shows useful logs on the website, but I guess the integration doesn't pass those data
![Screenshot 2022-02-07 at 22 16 30](https://user-images.githubusercontent.com/16367422/152873295-7250d3ac-ee42-4018-a43e-7d1a13e9d0f6.png)
.

**Comment by frenck** [comment]
> but I guess the integration doesn't pass those data

The Tuya API cloud doesn't pass that :) That is what the diagnostics shows. The data you see in the Tuya IoT debugger is not equals to what goes over the line (unfortunately).

**Comment by vadimbz** [comment]
I'm late for sad party, but can confirm that...
```
    "function": {},
    "status_range": {},
    "status": {},
```
...are empty. Kindly leave the issue open in case Tuya updates its API.

**Comment by alexanv1** [comment]
I opened an issue with Tuya and they were able to add status and instruction sets for ISV-200W. You should see the instruction\status sets in the Tuya IOT Portal now under the "Device Debugging" tab. Seems to work for my device which should now allow proper integration to HA.
If it still doesn't work for your device, you should be able to open an issue with Tuya and they should be able to resolve it.

**Comment by Zarkstar** [comment]
Mine updated automatically and now shows some useful info in the diagnostic report! I've attached it here
[tuya-4b3c280461170972843ad005306ffbb4-Sous Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json-2.txt](https://github.com/tuya/tuya-home-assistant/files/8336110/tuya-4b3c280461170972843ad005306ffbb4-Sous.Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json-2.txt)
.

**Comment by alexanv1** [comment]
Hmmm, it's interesting that the function\status_range sections are still empty. Possibly HA needs to be restarted for it to pick up the new data for those sections (I haven't restarted my instance yet either)

**Comment by Zarkstar** [comment]
You were right, it needed to be restarted!
[tuya-4b3c280461170972843ad005306ffbb4-Sous Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json-3.txt](https://github.com/tuya/tuya-home-assistant/files/8336236/tuya-4b3c280461170972843ad005306ffbb4-Sous.Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json-3.txt)

**Comment by Zarkstar** [comment]
[tuya-4b3c280461170972843ad005306ffbb4-Sous Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json-3.txt](https://github.com/tuya/tuya-home-assistant/files/8336250/tuya-4b3c280461170972843ad005306ffbb4-Sous.Vide-371c1fd1e10cbcc74bdd061ffb61a9e8.json-3.txt)
You were right! It needed to be restarted

**Comment by alexanv1** [comment]
I've modified the Tuya integration to support the device, will try to submit a PR to Home Assistant within the next day or two.
![sousvide](https://user-images.githubusercontent.com/44785744/159873940-5f264baf-1722-4698-8728-95ccec2e27fa.jpg)

**Comment by rodakswe** [comment]
> I've modified the Tuya integration to support the device, will try to submit a PR to Home Assistant within the next day or two. ![sousvide](https://user-images.githubusercontent.com/44785744/159873940-5f264baf-1722-4698-8728-95ccec2e27fa.jpg)

Hi, was this ever submitted and merged?
Best regards

**Comment by alexanv1** [comment]
Yeah - https://github.com/home-assistant/core/commit/0404c76c41bf4701e28cf761caf1c0cd5605dfdc
It's in as of HA 2202.7.0

**Comment by rodakswe** [comment]
> 

Perfect, totally missed it. 
Will setup tuya and try it out soon.
Thank you very much for your contribution 😁

**Comment by mksmo** [comment]
Hi!
I couldn't connect to Tuya integration - unsupported
[tuya-d9a88809fffeb969b0bd370ecf152830-ISV-100W2.0-ccf947348d4dd52e7cfe5d48266e5ea3.json.txt](https://github.com/tuya/tuya-home-assistant/files/9943605/tuya-d9a88809fffeb969b0bd370ecf152830-ISV-100W2.0-ccf947348d4dd52e7cfe5d48266e5ea3.json.txt)
<img width="378" alt="Снимок экрана 2022-11-05 в 14 02 39" src="https://user-images.githubusercontent.com/4104873/200116753-c1cb9619-1d24-49cd-9059-108c5d5006c1.png">
